### ISSUE TITLE
B033 does not behave the same as F601

### ISSUE THREAD

**Issue by drkspace** [astral-sh/ruff#26563]
### Summary

B033 (duplicate set values) does not catch the same duplicate entries as F601 (duplicate dict keys) for at least tuples.

For example, the set does not produce an error

```python
dict_ = {
    (1,): 1,
    (1,): 1
}

set_ = {
    (1,),
    (1,)
}
``` 

```sh
$ uvx ruff check --select B033,F601       
F601 Dictionary key literal `(1,)` repeated
 --> a.py:3:5
  |
1 | dict_ = {
2 |     (1,): 1,
3 |     (1,): 1
  |     ^^^^
4 | }
  |
help: Remove repeated key literal `(1,)`

Found 1 error.
No fixes available (1 hidden fix can be enabled with the `--unsafe-fixes` option).

```

[playground link](https://play.ruff.rs/13bdd067-76c7-453c-bd0a-67422b9f2e19)

(neither catch duplicate `NamedTuple` entries, but that will probably should be a separate issue)

### Version

ruff v0.15.20

**Comment by ntBre** [comment]
Thanks for the report! Yeah, it looks like `F601` has its own `is_literal_key` helper:

https://github.com/astral-sh/ruff/blob/e768389988a1beed3102bbe6df56a85d2b619bf0/crates/ruff_linter/src/rules/pyflakes/rules/repeated_keys.rs#L244

which is different from the more general `is_literal_expr` used by `B033`:

https://github.com/astral-sh/ruff/blob/e768389988a1beed3102bbe6df56a85d2b619bf0/crates/ruff_linter/src/rules/flake8_bugbear/rules/duplicate_value.rs#L76

I think it makes sense to make them consistent, although we might want to check the flake8-bugbear version of `B033` for parity.

I also think `is_literal_key` is actually a bit too aggressive. It considers any tuple to be a literal, even if it contains non-literal elements.

https://play.ruff.rs/f0f02aa0-cb17-4930-9700-2d676de284d4

We should be able to fix both of these in a shared helper.

**Comment by durationextender** [comment]
I would like to work on this issue @ntBre
